### PR TITLE
New extension Swiftlet

### PR DESCRIPTION
**Pull request by qwordy**
Resolve https://github.com/Azure/azure-cli/issues/14418

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review?thanks

File: src/swiftlet/azext_swiftlet/generated/_help.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      - name: Create a virtual machine with password authentication.
+        text: |-
+               az swiftlet virtual-machine create --location "westus" --password "{your-password}," --ports \
+port-range="3389" protocol="*" --startup-script "{inline startup script}" --swiftlet-bundle-sku "Windows_1" \
+--swiftlet-image-id "windows-2019-datacenter" --username "SwiftletUser" --tags key1="value1" key2="value2" \
+--resource-group "myResourceGroup" --vm-name "myVirtualMachine"
+      - name: Create a virtual machine with ssh authentication.
+        text: |-
+               az swiftlet virtual-machine create --location "westus" --ports port-range="22" protocol="*" \
+--ssh-public-key "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCeClRAk2ipUs/l5voIsDC5q9RI+YSRd1Bvd/O+axgY4WiBzG+4FwJWZm/mLLe5D\
```
Comment (yungezz) [inline src/swiftlet/azext_swiftlet/generated/_help.py:64]: how about use a fake key?

File: src/swiftlet/azext_swiftlet/generated/_help.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+               az swiftlet virtual-machine stop --resource-group "myResourceGroup" --vm-name "myVirtualMachine"
+"""
+
+helps['swiftlet virtual-machine wait'] = """
+    type: command
+    short-summary: Place the CLI in a waiting state until a condition of the swiftlet virtual-machine is met.
+    examples:
+      - name: Pause executing next line of CLI script until the swiftlet virtual-machine is successfully created.
+        text: |-
+               az swiftlet virtual-machine wait --resource-group "myResourceGroup" --vm-name "myVirtualMachine" \
```
Comment (yungezz) [inline src/swiftlet/azext_swiftlet/generated/_help.py:145]: vm?

File: src/swiftlet/azext_swiftlet/generated/_params.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azext_swiftlet.action import (
+    AddSwiftletVirtualMachineCreatePorts,
+    AddSwiftletVirtualMachineUpdatePorts
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('swiftlet virtual-machine list') as c:
```
Comment (yungezz) [inline src/swiftlet/azext_swiftlet/generated/_params.py:27]: vm?

File: src/swiftlet/azext_swiftlet/generated/_params.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                   'type is "linux", this is optional if sshPublicKey is set. If the image platform type is "windows", '
+                   'this is required.')
+        c.argument('ports', action=AddSwiftletVirtualMachineCreatePorts, nargs='*', help='The ports on which inbound '
+                   'traffic will be allowed.')
+        c.argument('startup_script', type=str, help='An inline script that will run upon startup of the virtual '
+                   'machine.')
+
+    with self.argument_context('swiftlet virtual-machine update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('vm_name', type=str, help='The name of the virtual machine.', id_part='name')
```
Comment (yungezz) [inline src/swiftlet/azext_swiftlet/generated/_params.py:57]: does it support id? name should be enough right, since subcommand is vm.

**Comment by qwordy** [comment]
Do not review generated code if I rewrite them in manual folder.

File: src/swiftlet/azext_swiftlet/generated/_help.py (hunk 1 above)
Comment (qwordy) [reply src/swiftlet/azext_swiftlet/generated/_help.py:64]: OK

File: src/swiftlet/azext_swiftlet/generated/_params.py (hunk 4 above)
Comment (qwordy) [reply src/swiftlet/azext_swiftlet/generated/_params.py:57]:
```
Resource Id Arguments
    --ids               : One or more resource IDs (space-delimited). It should be a complete
                          resource ID containing all information of 'Resource Id' arguments. You
                          should provide either --ids or other 'Resource Id' arguments.
    --name -n           : The name of the virtual machine.
    --resource-group -g : Name of resource group. You can configure the default group using `az
                          configure --defaults group=<name>`.
    --subscription      : Name or ID of subscription. You can configure the default subscription
                          using `az account set -s NAME_OR_ID`.
```

**Pull request merged**